src/main/java/in/muthukumari/validator/BankDetailValidator.java

```diff
@@ -23,18 +27,121 @@ public static boolean isValidAccountNumber(CustomerBankDetail customer) throws D
 
 		boolean isValidAccountNumber = false;
 		String accountNumberStr = Long.toString(customer.getAccountNumber());
-		int lengthOfAccountNumberStr=accountNumberStr.length();
+		int lengthOfAccountNumberStr = accountNumberStr.length();
 		Map<String, Integer> bankAndAccountNumberLengthList = AccountNumberLength.getAccountNumberLength();
 		if (bankAndAccountNumberLengthList.containsKey(customer.getBankName())) {
 			int accountNumberLength = bankAndAccountNumberLengthList.get(customer.getBankName());
-			if (accountNumberLength == lengthOfAccountNumberStr) {
+			if (accountNumberLength == lengthOfAccountNumberStr &&  accountNumberStr.matches(regex)) {
 				isValidAccountNumber = true;
 			}
-		}
-		else {
+		} else {
```

Comment:
DB exception only throw from D.A.O not throwed from validator

Reply:
I have changed this